lib/ecto/changeset.ex

```diff
@@ -2936,34 +2936,39 @@ defmodule Ecto.Changeset do
       when is_function(msg_func, 1) or is_function(msg_func, 3) do
     errors
     |> Enum.reverse()
-    |> merge_error_keys(msg_func, changeset)
-    |> merge_related_keys(changes, types, msg_func)
+    |> merge_keyword_keys(msg_func, changeset)
+    |> merge_related_error_keys(changes, types, msg_func)
```

Comment:
```suggestion
    |> merge_related_keys(changes, types, msg_func, &traverse_errors/2)
```
No need to hide this in a private function